### ISSUE TITLE
Could not find module `@babel/runtime/helpers/typeof/_index.mjs` imported from `@babel/runtime/helpers/esm/typeof`

### ISSUE THREAD

**Issue by anandasansol** [babel/babel#12914]
## Bug Report

- [ ] I would like to work on a fix!

**Current behavior**
I created an app with ember using npm and ember-cli ("ember-cli": "^2.18.2"). It was running very well from last 3 years. From last week facing issue while running the app.

**Input Code**
import _extends from `@babel/runtime/helpers/esm/typeof`;
results in this error:
Uncaught Error: Could not find module `@babel/runtime/helpers/typeof/_index.mjs` imported from `@babel/runtime/helpers/esm/typeof`

Calling any other modules available under /node_modules works fine.

**Expected behavior**
Need to be able to call @babel/runtime/...

**Babel Configuration (babel.config.js, .babelrc, package.json#babel, cli command, .eslintrc)**
- Filename: `package.json`

```json
{
  "name": "my-ember-ui",
  "version": "0.0.0",
  "private": true,
  "description": "Small description goes here",
  "license": "MIT",
  "author": "",
  "directories": {
    "doc": "doc",
    "test": "tests"
  },
  "repository": "",
  "scripts": {
    "build": "ember build",
    "start": "ember server",
    "test": "ember test"
  },
  "devDependencies": {
    "bootstrap": "^3.3.7",
    "broccoli-asset-rev": "^2.4.5",
    "broccoli-static-compiler": "^0.2.2",
    "ember-ajax": "^3.0.0",
    "ember-bootstrap": "^1.0.0",
    "ember-cli": "^2.18.2",
    "ember-cli-app-version": "^3.0.0",
    "ember-cli-babel": "^7.3.0-beta.1",
    "ember-cli-cors": "0.0.2",
    "ember-cli-dependency-checker": "^2.0.0",
    "ember-cli-eslint": "^4.0.0",
    "ember-cli-htmlbars": "^2.0.1",
    "ember-cli-htmlbars-inline-precompile": "^1.0.0",
    "ember-cli-inject-live-reload": "^1.4.1",
    "ember-cli-jstree": "^1.0.9",
    "ember-cli-notifications": "^4.3.3",
    "ember-cli-qunit": "^4.0.0",
    "ember-cli-shims": "^1.1.0",
    "ember-cli-uglify": "^1.2.0",
    "ember-concurrency": "^0.8.14",
    "ember-crumbly": "^1.0.7",
    "ember-data": "~2.15.0",
    "ember-export-application-global": "^2.0.0",
    "ember-file-upload": "^2.5.0",
    "ember-flatpickr": "^2.7.0",
    "ember-font-awesome": "4.0.0-rc.4",
    "ember-introjs": "2.4.0",
    "ember-light-table": "^1.12.2",
    "ember-load-initializers": "^1.0.0",
    "ember-models-table": "^2.0.0-beta.3",
    "ember-one-way-select": "^4.0.0",
    "ember-paper-stepper": "0.0.10",
    "ember-power-select": "^2.0.9",
    "ember-resolver": "^4.0.0",
    "ember-responsive": "^2.0.5",
    "ember-source": "~2.15.0",
    "ember-welcome-page": "^3.0.0",
    "loader.js": "^4.2.3"
  },
  "engines": {
    "node": "^4.5 || 6.* || >= 7.*"
  },
  "dependencies": {
    "@babel/runtime": "^7.9.0",
    "ember-cli-js-cookie": "^0.1.1",
    "ember-timepicker": "^0.3.0"
  }
}

```

**Environment**
Babel version(s): @babel/runtime 7.9.0
Node/npm version: Node v10.15.3
OS: Windows 10 Enterprise, 64 bit
```

```

**Possible Solution**

**Additional context**
Note: "@babel/runtime": "^7.9.0" in dependencies added for testing if this can resolve the issue. But unable to fix.
@babel/runtime/helpers/esm/typeof.js and @babel/runtime/helpers/typeof/_index.mjs exists under node_modules

**Comment by simoncoulton** [comment]
I've also just started seeing this today in our environment, the main differences between previously successful builds are the following:

`@babel/compat-data` 7.13.6 (was 7.13.0)
`@babel/parser` 7.13.4 (was 7.13.0)
`@babel/plugin-proposal-async-generator-functions` 7.13.5 (was 7.13.0)
`@babel/runtime` & `@babel/runtime-corejs3` 7.13.7 (was 7.13.1)

**Comment by nicolo-ribaudo** [comment]
I never used ember before and I don't know where to start, could one of you create a small repository that reproduces the issue? 🙏

**Comment by simoncoulton** [comment]
Just a quick update, if I pin `@babel/runtime` to 7.12.18 it works successfully. Just about to head off for the weekend, but i'll try to put together a small example for you when I'm back in the office on Monday 👍

**Comment by giano** [comment]
Same. App exploded

**Comment by simoncoulton** [comment]
@nicolo-ribaudo on further investigation I have determined that our `resolve.extensions` did not include `.mjs`. Adding that here has appeared to resolve (no pun intended) the issue. Appreciate your other PR for others though!

**Comment by giano** [comment]
TL:DR:
After "fixing" every package version to the last version I found on the working running instance I pinpointed that you can temporary fix the issue "fixing" "@babel/compat-data", "@babel/parser" and "@babel/runtime" to a previous version. The versions working for me are as such:
`
"@babel/cli": "7.12.17",
"@babel/compat-data": "7.12.13",
"@babel/core": "7.12.17",
"@babel/preset-env": "7.12.17",
"@babel/parser": "7.12.17",
"@babel/runtime": "7.12.18"
`
FULL WORKING PACKAGE.json FOR ME:

`{
    "name": "fair",
    "version": "1.1.0",
    "private": true,
    "description": "Final Analytical Interface & Repository",
    "repository": "",
    "license": "MIT",
    "author": "",
    "directories": {
        "doc": "doc",
        "test": "tests"
    },
    "scripts": {
        "build": "ember build",
        "clean-tmp": "ember clean-tmp",
        "start": "ember server",
        "test": "ember test"
    },
    "dependencies": {
        "ember-let": "0.7.1",
        "eslint": "6.8.0",
        "md5-dir": "0.2.0"
    },
    "devDependencies": {
        "@babel/cli": "7.12.17",
        "@babel/compat-data": "7.12.13",
        "@babel/core": "7.12.17",
        "@babel/preset-env": "7.12.17",
        "@babel/parser": "7.12.17",
        "@babel/runtime": "7.12.18",
        "body-parser": "1.17.2",
        "bootstrap": "4.0.0-alpha.6",
        "broccoli-asset-rev": "2.4.6",
        "ember-ajax": "3.0.0",
        "ember-api-actions": "git+https://git@github.com/giano/ember-api-actions.git",
        "ember-auto-import": "1.10.1",
        "ember-body-class": "1.1.5",
        "ember-bootstrap": "1.0.0-rc.3",
        "ember-bootstrap-cp-validations": "1.0.0",
        "ember-bootstrap-power-select": "1.1.0",
        "ember-c3": "1.3.1",
        "ember-clean-tmp": "0.1.4",
        "ember-cli": "2.18.2",
        "ember-cli-animate-css": "0.0.1",
        "ember-cli-app-version": "3.0.0",
        "ember-cli-babel": "7.23.1",
        "ember-cli-bootstrap-datepicker": "0.5.6",
        "ember-cli-content-security-policy": "0.6.2",
        "ember-cli-copyable": "0.9.6",
        "ember-cli-date-textbox": "1.2.0",
        "ember-cli-dependency-checker": "1.3.0",
        "ember-cli-deprecation-workflow": "0.2.5",
        "ember-cli-document-title": "0.3.3",
        "ember-cli-eslint": "3.0.3",
        "ember-cli-google-fonts": "2.14.1",
        "ember-cli-head": "0.2.2",
        "ember-cli-htmlbars": "4.3.1",
        "ember-cli-inject-live-reload": "2.0.2",
        "ember-cli-json-pretty": "0.1.4",
        "ember-cli-moment-range": "0.1.4",
        "ember-cli-moment-shim": "3.8.0",
        "ember-cli-qunit": "4.0.2",
        "ember-cli-sass": "6.2.0",
        "ember-cli-shims": "1.1.0",
        "ember-cli-sri": "2.1.1",
        "ember-cli-string-helpers": "1.4.0",
        "ember-cli-text-support-mixins": "1.1.2",
        "ember-cli-uglify": "3.0.0",
        "ember-component-css": "0.6.9",
        "ember-component-inbound-actions": "1.2.1",
        "ember-composable-helpers": "2.0.3",
        "ember-content-editable": "1.0.5",
        "ember-controller-lifecycle": "0.4.1",
        "ember-cp-validations": "3.5.6",
        "ember-cryptojs-shim": "4.0.0",
        "ember-data": "2.18.5",
        "ember-data-in-memory-adapter": "0.0.3",
        "ember-drag-drop": "0.7.0",
        "ember-font-awesome": "3.0.5",
        "ember-gestures": "0.4.8",
        "ember-get-config": "0.2.4",
        "ember-hammertime": "1.2.5",
        "ember-hold-button": "0.3.5",
        "ember-i18n": "5.0.2",
        "ember-i18n-cp-validations": "3.1.0",
        "ember-is-mobile": "2.0.5",
        "ember-link-action": "0.0.38",
        "ember-load-initializers": "1.0.0",
        "ember-local-storage": "1.3.7",
        "ember-lodash": "4.19.5",
        "ember-math-helpers": "2.1.1",
        "ember-moment": "8.0.1",
        "ember-notify": "5.2.1",
        "ember-parachute": "0.2.2",
        "ember-popper": "0.6.4",
        "ember-power-select": "2.3.5",
        "ember-power-select-blockless": "0.3.1",
        "ember-power-select-typeahead": "0.7.4",
        "ember-power-select-with-create": "0.6.2",
        "ember-promise-helpers": "1.0.9",
        "ember-radio-button": "1.1.1",
        "ember-resolver": "4.3.1",
        "ember-service-worker": "0.6.13",
        "ember-service-worker-asset-cache": "0.6.4",
        "ember-service-worker-cache-fallback": "0.6.2",
        "ember-service-worker-index": "0.6.9",
        "ember-simple-auth": "1.4.2",
        "ember-simple-auth-token": "2.1.0",
        "ember-sortable": "1.11.1",
        "ember-source": "2.13.4",
        "ember-sticky-element": "0.2.3",
        "ember-text-highlight": "1.2.3",
        "ember-toggle": "5.0.2",
        "ember-truth-helpers": "1.3.0",
        "ember-uploader": "2.0.0",
        "ember-vendor-split": "1.2.0",
        "ember-web-app": "2.3.1",
        "ember-welcome-page": "3.1.1",
        "ember-world-flags": "0.2.7",
        "ember-wormhole": "0.5.5",
        "eslint-config-prettier": "6.15.0",
        "eslint-plugin-prettier": "3.3.1",
        "glob": "4.5.3",
        "http-proxy": "1.16.2",
        "install": "0.12.2",
        "keyevent": "1.1.0",
        "loader.js": "4.6.0",
        "lodash-addons": "1.2.0",
        "mime": "2.5.2",
        "moment-timezone": "0.5.33",
        "morgan": "1.8.2",
        "popper.js": "1.16.1",
        "prettier": "1.19.1",
        "rate-limit-computed": "1.0.2",
        "sugar-date": "2.0.6",
        "torii": "0.9.6",
        "uglify-js": "3.9.1"
    },
    "engines": {
        "node": ">= 12",
        "yarn": ">= 1.21.1"
    }
}
`
